src/core/config/transformer/preset.rs

```diff
@@ -24,20 +30,15 @@ impl Transform for Preset {
         transform::default()
             .pipe(super::Required)
             .pipe(super::TreeShake.when(self.tree_shake))
-            .pipe(super::TypeMerger::new(self.merge_type))
+            .pipe(
+                super::TypeMerger::new(self.merge_type)
+                    .when(super::TypeMerger::is_enabled(self.merge_type)),
+            )
             .pipe(super::ImproveTypeNames.when(self.use_better_names))
-            .pipe(super::ConsolidateURL::new(self.consolidate_url))
+            .pipe(
+                super::ConsolidateURL::new(self.consolidate_url)
+                    .when(super::ConsolidateURL::is_enabled(self.consolidate_url)),
+            )
             .transform(config)
     }
 }
-
-impl Default for Preset {
-    fn default() -> Self {
-        Self {
-            merge_type: 1.0,
-            consolidate_url: 0.5,
-            use_better_names: true,
-            tree_shake: true,
-        }
-    }
-}
```

Comment:
Why do we need to delete this?

Reply:
I deleted it because the try_from implementation was using default to create a new instance of presets with default values. I don't think we need to remove it tbh because it's used in multiple tests, I reverted the default implementation and created a method on Preset instead to create an instance and used it in try_from.